tests/e2e/provider_journey_trace.py

```
@@ -0,0 +1,237 @@
+
+
+def _result_binding(tool_call_id: str, pointer: str) -> dict:
+    return {
+        "$trace_result": {
+            "tool_call_id": tool_call_id,
+            "pointer": pointer,
+        }
+    }
+
+
+def _normalize_google_arguments(trace: dict[str, Any], seeded_spreadsheet: str) -> None:
```

Ruff (0.16.0)

[warning] Too many branches (18 > 12)

(PLR0912)